### PR TITLE
Adding description to some of the v1beta3 objects.

### LINKED ISSUE #3628: Descriptions missing from v1beta3
Filing this as a remind to come back to it.

### PR DESCRIPTION
**Pull request by vishh**
Partially fixes #3628 

cc @bgrant0607

### CODE REVIEW

File: pkg/api/v1beta3/types.go [hunk 1]
```diff
@@ -595,48 +593,48 @@ type PodStatusResult struct {
 // to hosts.
 type Pod struct {
 	TypeMeta   `json:",inline"`
-	ObjectMeta `json:"metadata,omitempty"`
+	ObjectMeta `json:"metadata,omitempty" description:"metadata specific to the Pod"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Actually, all of the metadata fields provide identical information. The intention is that it is standardized across all kinds of objects.

```
description:"standard object metadata; see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md#metadata"
```

File: pkg/api/v1beta3/types.go [hunk 2]
```diff
@@ -595,48 +593,48 @@ type PodStatusResult struct {
...
 
 	// Spec defines the behavior of a pod.
-	Spec PodSpec `json:"spec,omitempty"`
+	Spec PodSpec `json:"spec,omitempty"description:"specification of the desired behavior of the pod"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Description is fine. Needs a space before "description".

Also add a link: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md#spec-and-status

File: pkg/api/v1beta3/types.go [hunk 3]
```diff
@@ -595,48 +593,48 @@ type PodStatusResult struct {
...
 
 	// Status represents the current information about a pod. This data may not be up
 	// to date.
-	Status PodStatus `json:"status,omitempty"`
+	Status PodStatus `json:"status,omitempty" description:"current status of the pod; populated by the system, read-only"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: "mostly recently observed status of the pod..."

Also please add the link: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md#spec-and-status

File: pkg/api/v1beta3/types.go [hunk 4]
```diff
@@ -595,48 +593,48 @@ type PodStatusResult struct {
...
 }
 
 // PodList is a list of Pods.
 type PodList struct {
 	TypeMeta `json:",inline"`
-	ListMeta `json:"metadata,omitempty"`
+	ListMeta `json:"metadata,omitempty" description:"metadata specific to PodList"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: standard list metadata; see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md#types-kinds

**Comment by bgrant0607** [comment]
I just commented on Pod and PodList, but please use the same pattern for all objects.

File: pkg/api/v1beta3/types.go (hunk 1 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Done

File: pkg/api/v1beta3/types.go (hunk 2 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Done

File: pkg/api/v1beta3/types.go (hunk 4 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Done

File: pkg/api/v1beta3/types.go (hunk 3 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Done

**Comment by vishh** [comment]
@bgrant0607: Addressed comments. PTAL.

File: pkg/api/v1beta3/types.go [hunk 5]
```diff
@@ -52,19 +52,19 @@ import (
...
-	APIVersion string `json:"apiVersion,omitempty"`
+	APIVersion string `json:"apiVersion,omitempty" description:"version of the schema the object should have"`
 }
 
 // ListMeta describes metadata that synthetic resources must have, including lists and
 // various status objects.
 type ListMeta struct {
 	// SelfLink is a URL representing this object.
-	SelfLink string `json:"selfLink,omitempty"`
+	SelfLink string `json:"selfLink,omitempty" description:"URL for the object"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: Add:
`; populated by the system, read-only`

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@vishh Please fix the ListMeta SelfLink description in your next PR. Also, the documentation of which fields cannot be updated has been merged, so please update the descriptions with that info.

File: pkg/api/v1beta3/types.go (hunk 5 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Fixed via #5150

**Pull request merged**